iceoryx_hoofs/testing/include/iceoryx_hoofs/testing/expect_no_death.hpp

```diff
@@ -25,9 +25,7 @@
     EXPECT_EXIT(                                                                                                       \
         {                                                                                                              \
             callable();                                                                                                \
-            /* Use fprintf here instead of std::cerr. cerr can be rerouted to another stream which can lead to failing \
-             * tests in combination with internal::CaptureStderr() for instance. */                                    \
-            fprintf(stderr, "callable did not terminate");                                                             \
+            IOX_LOG(ERROR) << "callable did not terminate";                                                            \
             exit(0);                                                                                                   \
```

Comment:
Please keep the old implementation. Since logger output is suppressed and since exit is called directly it will not be flushed to the console. It is okay for test code to use other means for output than the logger